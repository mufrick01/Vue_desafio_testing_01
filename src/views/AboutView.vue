<template>
    <h1>
        AboutView works
    </h1>
</template>

<script setup>

</script>
