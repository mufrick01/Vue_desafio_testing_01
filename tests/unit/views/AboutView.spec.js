import { mount } from '@vue/test-utils'
import { createRouter, createWebHistory } from 'vue-router'

import AboutView from '@/views/AboutView.vue'



describe('<AboutView/>', () => { 
    test('Probando la existencia del componente o vista AboutView ', async () => {
        const router = createRouter({
            history: createWebHistory(),
            routes: [{
                path: '/about',
                name: 'about',
                component: AboutView
            }],
        })
        router.push('/about')
        await router.isReady()

        const wrapper = mount(AboutView, {
            global: {
                plugins: [router]
            }
        })
        expect(wrapper.findComponent(AboutView).exists()).toBe(true)
    })
    
    test('AboutView deber√≠a tener un <h1> con el texto correcto',()=>{

        const wrapper = mount(AboutView);

        expect(wrapper.find('h1').text()).toBe('AboutView works');

    })

    test('el componente debe coincidir con el snapshot ', ()=>{
    

        const wrapper = mount(AboutView)

        expect(wrapper.html()).toMatchSnapshot()
    })
 })