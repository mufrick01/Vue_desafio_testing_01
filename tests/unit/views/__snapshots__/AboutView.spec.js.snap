// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutView/> el componente debe coincidir con el snapshot  1`] = `<h1> AboutView works </h1>`;
